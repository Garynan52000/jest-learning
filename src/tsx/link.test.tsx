import * as React from 'react';
import * as renderer from 'react-test-renderer';
import { Link } from './Link';

it('renders with no props', () => {
    const tree = renderer
        .create(<Link></Link>)
        .toJSON();
    expect(tree).toMatchSnapshot();
});


it('renders as FaceBook _blank link', () => {
    const tree = renderer
        .create(<Link href="http://www.facebook.com" target="_blank">Facebook</Link>)
        .toJSON();
    expect(tree).toMatchSnapshot();
});


it('properly escapes quotes', () => {
    const tree = renderer
        .create(<Link>{ "\"Facebook\" \\'is \\ 'awesome'" }</Link>)
        .toJSON();
    expect(tree).toMatchSnapshot();
});

it('changes the class when hovered', () => {
    const component = renderer.create(
        <Link>HOVER</Link>,
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();

    // manually trigger the callback
    renderer.act(() => {
        tree.props.onMouseEnter();
    });
    // re-rendering
    tree = component.toJSON();
    expect(tree).toMatchSnapshot();

    // manually trigger the callback
    renderer.act(() => {
        tree.props.onMouseLeave();
    });
    // re-rendering
    tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});