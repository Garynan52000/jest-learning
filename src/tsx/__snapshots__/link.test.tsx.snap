// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`changes the class when hovered 1`] = `
<a
  className="normal"
  href="javascript:;"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  target="_self"
>
  HOVER
</a>
`;

exports[`changes the class when hovered 2`] = `
<a
  className="hover"
  href="javascript:;"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  target="_self"
>
  HOVER
</a>
`;

exports[`changes the class when hovered 3`] = `
<a
  className="normal"
  href="javascript:;"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  target="_self"
>
  HOVER
</a>
`;

exports[`properly escapes quotes 1`] = `
<a
  className="normal"
  href="javascript:;"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  target="_self"
>
  "Facebook" \\'is \\ 'awesome'
</a>
`;

exports[`renders as FaceBook _blank link 1`] = `
<a
  className="normal"
  href="http://www.facebook.com"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  target="_blank"
>
  Facebook
</a>
`;

exports[`renders with no props 1`] = `
<a
  className="normal"
  href="javascript:;"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  target="_self"
>
  按钮
</a>
`;
